core/network/subnet.go

```diff
@@ -197,6 +197,27 @@ func (s SubnetInfos) GetByCIDR(cidr string) (SubnetInfos, error) {
 	return matching, nil
 }
 
+// Get by address returns subnets that based on IP range,
+// include the input IP address.
+func (s SubnetInfos) GetByAddress(addr string) (SubnetInfos, error) {
```

Comment:
It seems to me like this method could be used by the space lookup logic that takes care of the network info sent to the controller by the instance poller. Changing that piece of code to use this method might allow some code to be removed.

Reply:
Indeed. There are now several places where multiple DB hits might be ameliorated by using the growing domain tool-set here.